File: packages/snow_draw_engine/lib/draw/render/planning/watermark_visibility.dart
import '../../config/draw_config.dart';

/// Whether [config] produces visible watermark pixels.
bool isWatermarkVisible(WatermarkConfig config) {
  if (config.text.trim().isEmpty) {
    return false;
  }

  // At 8-bit precision an alpha below 1/255 ~= 0.004 maps to zero.
  return config.color.a * config.opacity >= 0.004;
}

File: packages/snow_draw_rust_engine/src/draw/render/planning/watermark_visibility.rs
#![allow(dead_code)]

use crate::draw::config::watermark_config::WatermarkConfig;

/// Returns whether watermark configuration produces visible pixels.
pub fn is_watermark_visible(config: &WatermarkConfig) -> bool {
    if config.text.trim().is_empty() {
        return false;
    }

    // At 8-bit precision an alpha below 1/255 ~= 0.004 maps to zero.
    config.color.a() * config.opacity >= 0.004
}
